include/sempr/entity/RDFValue.hpp

```diff
@@ -145,7 +160,7 @@ class RDFValue {
 
     /** Just another way to cast. */
     template <typename T>
-    T get() const {
+    T get() {
```

Comment:
Why not const?
Its hard to follow the implicit cast behind this. On the first view its like magic to make `*this` to the type `T`. Could you add a short brief to describe the behavior in behind?

Reply:
> Why not const?

Because this is just a different way to use the `operator T()`, which is called when trying to cast the RDFValue to something else, and the operator cannot be const (due to the makeValid, I think). I recognized the error when working on the issue, where one solution is to use `get<Sometype>();` when the cast would be ambiguous. So I fixed it and removed the `const`.

> Its hard to follow the implicit cast behind this. On the first view its like magic to make *this to the type T. Could you add a short brief to describe the behavior in behind?

Of course this is magic. The RDFValue is not intended to be used directly, but only in combination with the RDFPropertyMap. The user shall never actually create an instance of it. But he does get an RDFValue whenever he accesses the map, as it is the wrapper for any value and type -- a bit like a QVariant, as is described in the README.

The general thought behind this is:
`template <typename T> operator T()` is used to cast the RDFValue to the actual value the user wanted, and `template <typename T> RDFValue& operator = (const T& other)` allows to assign it _any_ value.

If the cast is ambiguous, use `template <typename T> T get();`

Is this explanaition sufficient? And if yes, which part of it is really missing? ^^

Reply:
Yes it is. It helps to point at the other operators that are used implicit. With this you can understand whats going on here with the magic. :-)

Reply:
I've added a few more lines to the readme. :)